core/trino-main/src/test/java/io/trino/memory/TestMemoryTracking.java

```diff
@@ -350,6 +350,22 @@ public void testDestroy()
         assertOperatorMemoryAllocations(operatorContext.getOperatorMemoryContext(), 0, 0, 0);
     }
 
+    @Test
+    public void testCumulativeUserMemoryEstimation()
+    {
+        LocalMemoryContext userMemory = operatorContext.localUserMemoryContext();
+        long userMemoryBytes = 100_000_000;
+        userMemory.setBytes(userMemoryBytes);
+        long startTime = System.nanoTime();
+        double cumulativeUserMemory = taskContext.getTaskStats().getCumulativeUserMemory();
+        long endTime = System.nanoTime();
+
+        double elapsedTimeInMillis = (endTime - startTime) / 1_000_000.0;
+        long averageMemoryForLastPeriod = userMemoryBytes / 2;
+
+        assertTrue(cumulativeUserMemory < elapsedTimeInMillis * averageMemoryForLastPeriod);
```

Comment:
avoid assertTrue with expressions, as the exc msg is going to be useless; use assertThat(...) instead

Reply:
@sopel39 it makes sense to me that `cumulativeUserMemory < elapsedTimeInMillis * averageMemoryForLastPeriod` especially when sleeping. But I'm not sure why we are dividing by two? I left that in since it's a tighter bound. I can look further if needed.

Reply:
`cumulativeUserMemory` is integral of memory usage over time, so it should be around `elapsedTimeInMillis * userMemoryBytes`, shouldn't it? Hence IMO it should be larger then `lapsedTimeInMillis * averageMemoryForLastPeriod` unless I don't understand something